app/page/info/tokens.tsx

```diff
@@ -1,84 +1,298 @@
 import { faCaretDown, faXmark } from "@fortawesome/free-solid-svg-icons"
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
 import * as ethers from "ethers"
-import { useState, type ChangeEvent, type FormEvent } from "react"
+import {
+  useCallback,
+  useEffect,
+  useState,
+  type ChangeEvent,
+  type FormEvent
+} from "react"
 
 import { useProviderContext } from "~app/context/provider"
 import { useAccount, useAction, useTokens } from "~app/storage"
-import { getChainName } from "~packages/network/util"
+import { getChainName, getErc20Contract } from "~packages/network/util"
+import address from "~packages/util/address"
+import number from "~packages/util/number"
 import type { BigNumberish, HexString } from "~typing"
 
 export function Tokens() {
   const tokens = useTokens()
   const account = useAccount()
-  const explorerUrl = `https://${getChainName(account.chainId)}.etherscan.io/`
+  const { provider } = useProviderContext()
+  const { updateToken } = useAction()
+
+  const [isTokenImportModalOpened, setIsTokenImportModalOpened] =
+    useState<boolean>(false)
+  const [selectedTokenAddress, setSelectedTokenAddress] =
+    useState<HexString>("")
+
+  const toggleTokenImportModal = useCallback(() => {
+    setIsTokenImportModalOpened((prev) => !prev)
+  }, [])
+  const openTokenInfoModal = useCallback((tokenAddress: HexString) => {
+    setSelectedTokenAddress(tokenAddress)
+  }, [])
+  const closeTokenInfoModal = useCallback(() => {
+    setSelectedTokenAddress("")
+  }, [])
+
+  useEffect(() => {
+    // TODO: In the future, adding an Indexer to the Background Script to
+    // monitor Account-related transactions. Updates like balance will trigger
+    // as needed, avoiding fixed interval polling with setInterval().
+    const updateTokenBalances = async () => {
+      await Promise.all(
+        tokens.map(async (token) => {
+          const balance = await getErc20Contract(
+            token.address,
+            provider
+          ).balanceOf(account.address)
+          if (
+            account.tokens.find((t) => t.address === token.address)?.balance !==
+            balance
+          ) {
```

Comment:
I think we can directly check `token.balance !== balance` here. And be aware that returned balance from contract would be `bigint`